mirgecom/flux.py

```diff
@@ -62,15 +150,61 @@ def lfr_flux(cv_tpair, f_tpair, normal, lam):
         object array of :class:`meshmode.dof_array.DOFArray` with outward-pointing
         normals
 
-    lam: :class:`meshmode.dof_array.DOFArray`
+    lam: :class:`~meshmode.dof_array.DOFArray`
+
+        lambda parameter for Lax-Friedrichs/Rusanov flux
+
+    Returns
+    -------
+    numpy.ndarray
+
+        object array of :class:`~meshmode.dof_array.DOFArray` with the
+        Lax-Friedrichs/Rusanov flux.
+    """
+    return flux_lfr(cv_tpair, f_tpair, normal, lam)@normal
+
+
+def flux_lfr(cv_tpair, f_tpair, normal, lam):
+    r"""Compute Lax-Friedrichs/Rusanov flux after [Hesthaven_2008]_, Section 6.6.
+
+    The Lax-Friedrichs/Rusanov flux is calculated as:
+
+    .. math::
+
+        f_{\mathtt{LFR}} = \frac{1}{2}(\mathbf{F}(q^-) + \mathbf{F}(q^+))
+        + \frac{\lambda}{2}(q^{-} - q^{+})\hat{\mathbf{n}},
+
+    where $q^-, q^+$ are the scalar solution components on the interior and the
+    exterior of the face on which the LFR flux is to be calculated, $\mathbf{F}$ is
+    the vector flux function, $\hat{\mathbf{n}}$ is the face normal, and $\lambda$
+    is the user-supplied jump term coefficient.
+
+    Parameters
+    ----------
+    cv_tpair: :class:`~grudge.trace_pair.TracePair`
+
+        Solution trace pair for faces for which numerical flux is to be calculated
+
+    f_tpair: :class:`~grudge.trace_pair.TracePair`
+
+        Physical flux trace pair on faces on which numerical flux is to be calculated
+
+    normal: numpy.ndarray
+
+        object array of :class:`~meshmode.dof_array.DOFArray` with outward-pointing
+        normals
+
+    lam: :class:`~meshmode.dof_array.DOFArray`
 
         lambda parameter for Lax-Friedrichs/Rusanov flux
```

Comment:
`lam` is a local/global upper bound on the wave speed. The docs should say so, here and elsewhere.

Reply:
The reason I let it be a passed-in parameter is because the LFR reads that lambda = the max eigenvalue of the system flux jacobian. For single gas/fluid, it is the max wave speed, but not for other systems and maybe not even for our system with species diffusion.  

See if this added documentation clears it up. (2ec3cc0)